### PR TITLE
Auto-generate IAM policies

### LINKED ISSUE #2269: Managed IAM Policy `AWSCloudWatchLambdaInsightsExecutionRolePolicy` incorrect ARN
### What happened?

According to this older issue https://github.com/pulumi/pulumi-aws/issues/1700 the `CloudWatchLambdaInsightsExecutionRolePolicy` was added as a constant in the AWS IAM managed policies list.  

However, the item is not within the typescript defined list and typing the role that was added in the referred Github issue (and related MR) gives me back an error that the role does not exist. My best guess is that the previous fix forgot a list variable or something.

### Steps to reproduce

```
new aws.iam.RolePolicyAttachment("lambda_role-lambda_insights-attachment", {
  role: signAndVerifyDeclarationsLambdaRole,
  policyArn: aws.iam.ManagedPolicies.AWSCloudWatchLambdaInsightsExecutionRolePolicy, // <-- this should return  "arn:aws:iam::aws:policy/AWSCloudWatchLambdaInsightsExecutionRolePolicy", but errors out instead.
});
```

### Expected Behavior

No error when you use the IAM managed policy `aws.iam.ManagedPolicies.AWSCloudWatchLambdaInsightsExecutionRolePolicy`.

### Actual Behavior

Error occurs.

### Output of `pulumi about`

```
Dependencies:
NAME                VERSION
@pulumi/aws         5.23.0
@pulumi/pulumi      3.49.0
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by danielrbradley**
Manually merge generated definitions with existing code.

Fixes #2269 
Superceeds #2384

### CODE REVIEW

**Comment by danielrbradley** [comment]
Ok, we've got a silly problem. Python adds `_` characters in the name, but some policy names only vary by a `_` e.g. 
```go
{Name: "AWSCloudTrail_FullAccess", Value: "arn:aws:iam::aws:policy/AWSCloudTrail_FullAccess"},
{Name: "AWSCloudTrailFullAccess", Value: "arn:aws:iam::aws:policy/AWSCloudTrailFullAccess"},
```

**Comment by danielrbradley** [comment]
Logged an issue around the python conflicts (https://github.com/pulumi/pulumi/issues/12314) and am just using the approach used elsewhere in the enum - to drop the leading "AWS" or "Amazon" on the new names.

File: docs/generate-iam-policies.jq [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/jq -Mfr
```
Comment (thomas11) [inline docs/generate-iam-policies.jq:1]: Prefer long arguments for committed files for readability

**Review by thomas11 — APPROVED** [review]
The next step would be to hook this up with CI so we're alerted to new policies.

File: docs/generate-iam-policies.jq (hunk 1 above)
Comment (danielrbradley) [reply docs/generate-iam-policies.jq:1]: Not actually using this method of invoking as it didn't work .. I'll just remove it

**Pull request merged**